### PR TITLE
test: improve teardown time

### PR DESCRIPTION
**Pull request by kimfy**
# test: improve teardown time

## Description

This PR aims to explore possibilities in reducing teardown time in automated tests. The biggest issue we're facing is removing role_definitions using Terraform. This PR solves this in two ways:
- Custom tests: by deleting the two provisioned role definitions using az cli
- Terraform test framework: by removing the testing of role_definitions all together

Test times (before/after this PR):
- Custom tests: 10-20 minutes to ~3 minutes
- Terraform test framework: ~10 minutes to ~3 minutes

Fixes: no issue on this

## Type of Change

- [ ] Bug fix
- [ ] New feature
- [ ] Documentation update
- [x] Other

## Testing
See the testing [docs](../tests/readme.md) for more information about how to test your code 
- [x] Performed a local `terraform plan`, `apply`, and `destroy` to validate changes.
- [x] All tests are passing.
- [x] Added, updated, or removed tests when appropriate

## Checklist

Before submitting this PR, please ensure the following:

- [x] My commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) style guide
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings. I have ran `terraform validate` in the root module and all sub modules where I have made changes

## Additional Information

Provide any additional information or context about the pull request here.

This PR introduces a tradeoff: we no longer test role definitions using the official test framework by Terraform because we can not control the destroy operation of this resource (which is what reduces test times).

---

Thank you for contributing to Station!

### CODE REVIEW

**Comment by kimfy** [comment]
@SanderBlom this is now ready for review 👍🏻

File: test/test_applications.tf [hunk 1]
```diff
@@ -1,99 +1,99 @@
...
-  }
-}
-
-resource "random_uuid" "maximum" {
-}
-
-resource "random_uuid" "maximum2" {
-}
-
+#module "station-applications" {
```
Comment (SanderBlom) [inline test/test_applications.tf]: Why is this test commented out?

File: test/test_groups.tf [hunk 2]
```diff
@@ -1,37 +1,37 @@
...
-      owners           = [data.azuread_client_config.current.object_id]
-      dynamic_membership = {
-        enabled = true
-        rule    = "user.jobTitle -eq \"DevOps Engineer\""
-      }
-    }
-  }
-}
-
+#module "station-groups" {
```
Comment (SanderBlom) [inline test/test_groups.tf]: Should this also be commented out?

File: test/test_user_assigned_identities.tf [hunk 3]
```diff
@@ -1,32 +1,32 @@
...
-        subscription_reader = {
-          role_definition_name = "Reader"
-          scope                = "/subscriptions/${data.azurerm_client_config.current.subscription_id}"
-        }
-      }
-    }
-  }
-}
-
+#module "station-uai" {
```
Comment (SanderBlom) [inline test/test_user_assigned_identities.tf]: Should this also be commented out?

**Review by SanderBlom — CHANGES_REQUESTED** [review]

File: test/test_applications.tf (hunk 1 above)
Comment (kimfy) [reply test/test_applications.tf]: Woops. No, this is an error - I disabled them while testing the role_definition teardowns. Fixed now

File: test/test_groups.tf (hunk 2 above)
Comment (kimfy) [reply test/test_groups.tf]: Woops. No, this is an error - I disabled them while testing the role_definition teardowns. Fixed now

File: test/test_user_assigned_identities.tf (hunk 3 above)
Comment (kimfy) [reply test/test_user_assigned_identities.tf]: Woops. No, this is an error - I disabled them while testing the role_definition teardowns. Fixed now

**Review by SanderBlom — APPROVED** [review]

**Pull request merged**